### PR TITLE
Bug 1162213 - [gaia-list] lint fix and add package.json for build and…

### PR DESCRIPTION
**Pull request by gasolin**
… travis test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,3 +9,86 @@ $ bower install gaia-components/gaia-list
 ## Examples
 
 - [Example](http://gaia-components.github.io/gaia-list/)
+
+## Usage
```
Comment (gasolin) [inline README.md]: There's room for syntax improvement but I'd rather go through other patch to fix it. Just document what we got now.

File: README.md [hunk 2]
```diff
@@ -9,3 +9,86 @@ $ bower install gaia-components/gaia-list
...
+	<i data-icon="forward-light"></i>
+  </a>
+</gaia-list>
+```
+
+List with switch and divider
+
+```
+<style>
+    gaia-list .divided {
```
Comment (gasolin) [inline README.md]: this visual hack does not match UX requirement, but lets go for another patch

File: README.md [hunk 3]
```diff
@@ -9,3 +9,86 @@ $ bower install gaia-components/gaia-list
...
 
 - [Example](http://gaia-components.github.io/gaia-list/)
+
+## Usage
+
+Settings style list
+
+```
+<gaia-list>
+  <a>
```
Comment (gasolin) [inline README.md]: To imply proper semantics, every row should be wrap in `<li>`, we could improve that in another bug

File: README.md [hunk 4]
```diff
@@ -9,3 +9,86 @@ $ bower install gaia-components/gaia-list
...
+    border-color:
+      var(--border-color,
+      var(--background-plus));
+  }
+ </style>
+
+
+<gaia-list>
+  <li class="ripple">
+    <label flex flexbox for="wifi-switch">
```
Comment (gasolin) [inline README.md]: looks hacky to add `flex flexbox`

Comment (wilsonpage) [inline README.md]: They do different things. `flex` expands element to fill available space `flex: 1`, `flexbox` defines the element as a `display: flex`.

**Comment by wilsonpage** [comment]
Looks good! I spotted a visual bug:

screen shot 2015-05-07 at 11 24 31

 The border on the left should be a vertical line, not rounded. This may be nothing to do with your patch, but thought I'd raise it.

**Comment by gasolin** [comment]
mmm... It because we set `divided` class in `<gaia-switch class="divided"></gaia-switch>`, so the  `divided` inherit gaia-switch's `border-radius: 18px`. add

```
::content .divided {
      border-radius: 0 !important;
```

could quick fix it, but !important is a bed smell though..

Consider https://bugzilla.mozilla.org/show_bug.cgi?id=1160964 ,
UX request if the list has a separator, the animation should not cross the separator, so we have to reconsider how we handle divider now.

File: index.html [hunk 5]
```diff
@@ -23,13 +25,9 @@
       color: var(--text-color-minus);
     }
 
+    /* XXX temporary hack to make divider not inherit from switch */
     gaia-list .divided {
-      border-left: solid 1px;
```
Comment (gasolin) [inline index.html]: general rules are located in gaia-list, add divider hack to overwrite border-round style inherit from `gaia-switch`
